source/docs/software/troubleshooting/robots-shouldnt-quit.rst

```diff
@@ -0,0 +1,415 @@
+
+A functional implementation could look like this: 
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. code-block:: Java
+          :lineno-start: 19
+
+            PWMSparkMax armMotorCtrl;
+
+            @Override
+            public void robotInit() {
+                
+                armMotorCtrl = new PWMSparkMax(0);
+                armMotorCtrl.setInverted(true);
+
+            }
+
+   .. group-tab:: C++
+
+      .. code-block:: C++
+
+         //TODO
+
+
+Divide by Zero
+^^^^^^^^^^^^^^
+
+It is not generally possible to divide an integer by zero, and expect reasonable results. Most processors (including the RoboRIO) will cause an Unhandled Exception.
+
+For example, consider the following code:
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. code-block:: Java
+          :lineno-start: 18
+
+            int armLengthRatio;
+            int elbowToWrist_in = 39;
+            int shoulderToElbow_in;
+
+            @Override
+            public void robotInit() {
+
+               armLengthRatio = elbowToWrist_in / shoulderToElbow_in;
+
+            }
+
+   .. group-tab:: C++
+
+      .. code-block:: C++
+
+         //TODO
+
+When run, you'll see output that looks like this:
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. code-block:: text
+
+         ********** Robot program starting **********
+         Error at frc.robot.Robot.robotInit(Robot.java:25): Unhandled exception: java.lang.ArithmeticException: / by zero
+               at frc.robot.Robot.robotInit(Robot.java:25)
+               at edu.wpi.first.wpilibj.TimedRobot.startCompetition(TimedRobot.java:94)
+               at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:335)
+               at edu.wpi.first.wpilibj.RobotBase.lambda$startRobot$0(RobotBase.java:387)
+               at java.base/java.lang.Thread.run(Thread.java:834)
+
+         Warning at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:350): Robots should not quit, but yours did!
+         Error at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:352): The startCompetition() method (or methods called by it) should have handled the exception above.
+
+   .. group-tab:: C++
+
+      .. code-block:: text
+
+          TODO
+
+Looking at the stack trace, we can see a ``java.lang.ArithmeticException: / by zero`` exception has occurred on line 25. If you look at the two variables which are used on the right-hand side of the ``=`` operator, you might notice one of them has not been initialized. This means its value is, by default, zero. And, the zero-value variable is used in the denominator of a division operation. Hence, the divide by zero error happens.
+
+Alternatively, by running the single-step debugger and stopping on line 25, you could inspect the value of all variables to discover ``shoulderToElbow_in`` has a value of ``0``.
+
+Fixing Div/0 Issues

```

Comment:
Unnecessary abbreviation?

Reply:
Possibly. Lemme go check the exact text of what gets printed out in C++/Java statements. The only goal of using an alternate way to "spell" the same concept is to enhance searchability.

Reply:
Ah okay. If it is using the same phrasing as the error message, then maybe consider putting the "Div/0" in a code block or italics.